app/models/metric/ci_mixin/capture.rb

```diff
@@ -33,25 +34,26 @@ def perf_capture(interval_name, start_time = nil, end_time = nil)
       return
     end
 
+    metrics_capture = perf_capture_object
     start_time, end_time = fix_capture_start_end_time(interval_name, start_time, end_time)
-    start_range, end_range, counters_data = just_perf_capture(interval_name, start_time, end_time)
+    start_range, end_range, counters_data = just_perf_capture(interval_name, start_time, end_time, metrics_capture)
 
     perf_process(interval_name, start_range, end_range, counters_data) if start_range
   end
 
+  # Determine the start_time for capturing if not provided
+  # interval_name == realtime is the only one that passes no start_time/end_time
   def fix_capture_start_end_time(interval_name, start_time, end_time)
-    start_time = start_time.utc unless start_time.nil?
-    end_time = end_time.utc unless end_time.nil?
-
-    # Determine the start_time for capturing if not provided
-    if interval_name == 'historical'
-      start_time ||= Metric::Capture.historical_start_time
-    elsif interval_name == "hourly"
-      start_time ||= last_perf_capture_on || 4.hours.ago.utc
-    else # interval_name == realtime
-      start_time ||= [last_perf_capture_on, 4.hours.ago.utc.beginning_of_day].compact.max
-    end
-    [start_time, end_time]
+    start_time ||=
+      case interval_name
+      when "historical" # Vm or Host
+        Metric::Capture.historical_start_time
+      when "hourly" # Storage (value is ignored)
+        last_perf_capture_on || 4.hours.ago.utc
+      else # "realtime" for Vm or Host
+        [last_perf_capture_on, 4.hours.ago.utc].compact.max
```

Comment:
This is a big difference from `[last_perf_capture_on, 4.hours.ago.utc.beginning_of_day]` to `[last_perf_capture_on, 4.hours.ago.utc]` (one we want to make just calling it out since it looks like a simple move from if/elsif -> case/when)

@kbrock can you find the PR that changed to add the `.beginning_of_day`?  I think it was "recent" right?

Reply:
Yea, I put that thread into the commit message:

run realtime metrics for past 4 hours

>This is probably the value we started with.

>We changed this But we were ending up with so many duplicate realtime metrics values on the queue
>that we were optimizing the values for removing duplicates rather than optimizing
>for the correct value, or a value that business wise makes sense

>issue was highlighted: https://github.com/ManageIQ/manageiq/pull/19838

>It was probably introduced due to confusion between realtime start and realtime start cutoff
>the latter being the cutoff for historicals sent in the past due to a realtime value going onto queue